bvm/ballerina-core/src/main/java/org/ballerinalang/util/cli/ArgumentParser.java

```diff
@@ -0,0 +1,461 @@
+                try {
+                    return JSONUtils.convertJSONToBArray(JsonParser.parse(value), (BArrayType) type);
+                } catch (BallerinaException e) {
+                    throw new BLangUsageException("invalid argument '" + value + "', expected array elements of "
+                                                          + "type: " + ((BArrayType) type).getElementType());
+                }
+            case TypeTags.MAP_TAG:
+                try {
+                    return JSONUtils.jsonToBMap(JsonParser.parse(value), (BMapType) type);
+                } catch (BallerinaException e) {
+                    throw new BLangUsageException("invalid argument '" + value + "', expected map argument of element"
+                                                          + " type: " + ((BMapType) type).getConstrainedType());
+                }
+            case TypeTags.UNION_TAG:
+                return parseUnionArg((BUnionType) type, value);
+            case TypeTags.TYPEDESC_TAG:
+                try {
+                    return new BTypeDescValue(getTypeFromName(value));
+                } catch (IllegalStateException e) {
+                    throw new BLangUsageException("invalid argument '" + value + "', unsupported/unknown typedesc "
+                                                          + "expected with entry function");
+                }
+            default:
+                throw new BLangUsageException(UNSUPPORTED_TYPE_PREFIX + " '" + type + "'");
+        }
+    }
+
+    private static long getIntegerValue(String argument) {
+        try {
+            if (argument.toUpperCase().startsWith(BINARY_PREFIX)) {
+                return Long.parseLong(argument.toUpperCase().replace(BINARY_PREFIX, ""), 2);
+            } else if (argument.toUpperCase().startsWith(HEX_PREFIX)) {
+                return Long.parseLong(argument.toUpperCase().replace(HEX_PREFIX, ""), 16);
+            }
+            return Long.parseLong(argument);
+        } catch (NumberFormatException e) {
+            throw new BLangUsageException("invalid argument '" + argument + "', expected integer value");
+        }
+    }
+
+    private static double getFloatValue(String argument) {
+        try {
+            return Double.parseDouble(argument);
+        } catch (NumberFormatException e) {
+            throw new BLangUsageException("invalid argument '" + argument + "', expected float value");
+        }
+    }
+
+    private static boolean getBooleanValue(String argument) {
+        if (!TRUE.equalsIgnoreCase(argument) && !FALSE.equalsIgnoreCase(argument)) {
+            throw new BLangUsageException("invalid argument '" + argument + "', expected boolean value 'true' or "
+                                                  + "'false'");
+        }
+        return Boolean.parseBoolean(argument);
+    }
+
+    private static byte getByteValue(String argument) {
+        long longValue; // TODO: 7/4/18 Allow byte literals?
```

Comment:
We should be able to support byte[] literals. Shall we create an issue for that. ?

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/10266, thanks!